src/api/categoryAPI.ts

```diff
@@ -0,0 +1,40 @@
+import { CategoryPayload } from "types/category";
+import { convertCamelCaseToSnakeCase } from "utils/convertObject";
+import api from ".";
+
+const categoryAPI = {
+  fetchCategoriesList: async (offset: number, limit: number) => {
+    return await api.request({
+      url: "/categories",
+      params: {
+        offset,
+        limit,
+      },
+    });
+  },
```

Comment:
No need define async await here. api.request is async function so action return value is also promise.

Reply:
Please check other places too.